test/open-uri/test_open-uri.rb

```diff
@@ -115,6 +116,13 @@ def test_open_too_many_arg
     assert_raise(ArgumentError) { open("http://192.0.2.1/tma", "r", 0666, :extra) {} }
   end
 
+  def test_tempfile_rest_parameter
+    temp_file = Tempfile.new
+    assert_nothing_raised { open(temp_file, 'a') }
```

Comment:
Wouldn't that mean that the usage of e.g. `open(temp_file, 'a')` would also leak the file descriptor in regular ruby programs? Maybe we should use another implementation instead of this. Have you another approach @nobu ?